features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/RumSessionEndedMetricTest.kt

```diff
@@ -0,0 +1,412 @@
+        val duration1 = TimeUnit.SECONDS.toNanos(15)
+        val duration2 = TimeUnit.SECONDS.toNanos(10)
+        val date1 = TimeUnit.SECONDS.toMillis(20)
+        val date2 = TimeUnit.SECONDS.toMillis(30)
+
+        val currentSessionViewEvent = event1.copy(
+            session = event1.session.copy(id = fakeSessionId),
+            date = date1,
+            view = event1.view.copy(timeSpent = duration1)
+        )
+        val differentSessionViewEvent = event2.copy(
+            date = date2,
+            view = event2.view.copy(timeSpent = duration2)
+        )
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        sessionEndedMetric.onViewTracked(currentSessionViewEvent)
+        sessionEndedMetric.onViewTracked(differentSessionViewEvent)
+
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes()
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val viewCounts = rse[SessionEndedMetric.VIEW_COUNTS_KEY] as Map<*, *>
+        assertEquals(1, viewCounts[SessionEndedMetric.VIEW_COUNTS_TOTAL_KEY])
+    }
+
+    @Test
+    fun `M view counts total correct W track views`(@Forgery events: List<ViewEvent>) {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+
+        // When
+        events.map {
+            it.copy(session = it.session.copy(id = fakeSessionId))
+        }.forEach {
+            sessionEndedMetric.onViewTracked(it)
+        }
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes()
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val viewCounts = rse[SessionEndedMetric.VIEW_COUNTS_KEY] as Map<*, *>
+        assertEquals(events.count(), viewCounts[SessionEndedMetric.VIEW_COUNTS_TOTAL_KEY])
+    }
+
+    @Test
+    fun `M view counts background correct W track views`(@Forgery events: List<ViewEvent>) {
+        // Given
+        val sessionEndedMetric = SessionEndedMetric(fakeSessionId, fakeStartReason)
+        val backgroundViewCount = Random.nextInt(events.size)
+
+        // When
+        events.mapIndexed { index, viewEvent ->
+            val url = if (index in 0 until backgroundViewCount) {
+                RumViewManagerScope.RUM_BACKGROUND_VIEW_URL
+            } else {
+                viewEvent.view.url
+            }
+            viewEvent.copy(
+                session = viewEvent.session.copy(id = fakeSessionId),
+                view = viewEvent.view.copy(url = url)
+            )
+        }.forEach {
+            sessionEndedMetric.onViewTracked(it)
+        }
+        // Then
+        val attributes = sessionEndedMetric.toMetricAttributes()
+        val rse = attributes[SessionEndedMetric.RSE_KEY] as Map<*, *>
+        val viewCounts = rse[SessionEndedMetric.VIEW_COUNTS_KEY] as Map<*, *>
+        assertEquals(backgroundViewCount, viewCounts[SessionEndedMetric.VIEW_COUNTS_BG_KEY])
```

Comment:
```suggestion
        assertThat(viewCounts[SessionEndedMetric.VIEW_COUNTS_BG_KEY]).hasSize(backgroundViewCount)
```

Reply:
Same, viewCounts[SessionEndedMetric.VIEW_COUNTS_TOTAL_KEY] returns actually an Int not a collection, so I don't think it has hasSize API, I updated with .isEqualTo()